File: pyjl/cases/langcomp_bench.py
#!/usr/bin/env python3

from typing import List


def test_python(iterations: int):
    iteration = 0
    total = float(0.0)
    array_length = 1000
    array: List[int] = [i for i in range(array_length)]
    print("iterations", iterations)
    while iteration < iterations:
        innerloop = 0
        while innerloop < 100:
            total += array[(iteration + innerloop) % array_length]
            innerloop += 1
        iteration += 1
    if total == 15150:
        print("OK")
    del array


if __name__ == "__main__":
    test_python(3)

File: pyjl/tests/tests_py2many-py2many/langcomp_bench.jl

function test_python(iterations::Int64)
iteration = 0
total = float(0.0)
array_length = 1000
array::Vector{Int64} = [i for i in (0:array_length - 1)]
while iteration < iterations
innerloop = 0
while innerloop < 100
total += array[((iteration + innerloop) % array_length) + 1]
innerloop += 1
end
iteration += 1
end
@assert(total == 15150)
println("OK");
empty!(array)
end

function main()
test_python(3);
end

main()
